client/src/context.rs

```diff
@@ -160,67 +143,44 @@ impl rpc::Server for DeviceCollectionServer {
                     (dcb.cb, dcb.user_ptr)
                 };
 
-                self.cpu_pool.spawn_fn(move || {
-                    run_in_callback(|| {
-                        if devtype.contains(cubeb_backend::DeviceType::INPUT) {
-                            unsafe {
-                                input_cb.unwrap()(ptr::null_mut(), input_user_ptr as *mut c_void)
-                            }
-                        }
-                        if devtype.contains(cubeb_backend::DeviceType::OUTPUT) {
-                            unsafe {
-                                output_cb.unwrap()(ptr::null_mut(), output_user_ptr as *mut c_void)
-                            }
+                run_in_callback(|| {
+                    if devtype.contains(cubeb_backend::DeviceType::INPUT) {
+                        unsafe { input_cb.unwrap()(ptr::null_mut(), input_user_ptr as *mut c_void) }
+                    }
+                    if devtype.contains(cubeb_backend::DeviceType::OUTPUT) {
+                        unsafe {
+                            output_cb.unwrap()(ptr::null_mut(), output_user_ptr as *mut c_void)
                         }
-                    });
+                    }
+                });
 
-                    Ok(DeviceCollectionResp::DeviceChange)
-                })
+                DeviceCollectionResp::DeviceChange
             }
         }
     }
 }
 
 impl ContextOps for ClientContext {
     fn init(_context_name: Option<&CStr>) -> Result<Context> {
-        fn bind_and_send_client(
-            stream: audioipc::AsyncMessageStream,
-            tx_rpc: &mpsc::Sender<rpc::ClientProxy<ServerMessage, ClientMessage>>,
-        ) {
-            let transport = framed(stream, Default::default());
-            let rpc = rpc::bind_client::<CubebClient>(transport);
-            // If send fails then the rx end has closed
-            // which is unlikely here.
-            let _ = tx_rpc.send(rpc);
-        }
-
         assert_not_in_callback();
 
-        let (tx_rpc, rx_rpc) = mpsc::channel();
-
         let params = AUDIOIPC_INIT_PARAMS.with(|p| p.replace(None).unwrap());
         let thread_create_callback = params.thread_create_callback;
         let thread_destroy_callback = params.thread_destroy_callback;
 
-        let server_stream =
-            unsafe { audioipc::MessageStream::from_raw_handle(params.server_connection) };
+        let server_stream = unsafe { sys::Pipe::from_raw_handle(params.server_connection) };
```

Comment:
I am not sure why this is named as `server_stream`. `ServerStream` is the cubeb stream on the ipc server-side/parent process. This should be the handle of `client_pipe` created in ipc server-side. Could we name this `server_connection`?

Reply:
Done.